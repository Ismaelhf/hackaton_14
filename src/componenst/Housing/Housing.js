import React from 'react';
import {
  StyleSheet,
  View,
  Image,
  TouchableHighlight,
  TouchableOpacity,
  Text,
  ScrollView,
} from 'react-native';

const Housing = ({data, navigation}) => {
  return (
    <>
      {data.map((item) => (
        <TouchableOpacity
          key={item.id}
          onPress={() => navigation.navigate('Housing', {item: {item}})}>
          <View>
            <Image
              style={styles.mejores}
              source={{
                uri: item.image,
              }}
            />
          </View>
        </TouchableOpacity>
      ))}
    </>
  );
};

const styles = StyleSheet.create({
  mejores: {
    width: '100%',
    height: 200,
    marginVertical: 5,
  },
});
export default Housing;
