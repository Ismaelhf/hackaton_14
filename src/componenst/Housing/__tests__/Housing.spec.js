// import { shallow, mount, render } from 'enzyme';
import React from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {shallow} from 'enzyme';
import Housing from '../Housing';
import {housingData} from '../../../data/housingData';

const initialProps = {
  data: housingData,
};

// console.log('test');
const wrapper = shallow(<Housing {...initialProps} />);

describe('Housing component', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should Housing element in component', () => {
    expect(wrapper.find(TouchableOpacity).length).toBe(3);
    expect(wrapper.find(View).length).toBe(3);
    expect(wrapper.find(Image).length).toBe(3);
  });
  it('should match props', () => {
    expect(wrapper.props.data).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
