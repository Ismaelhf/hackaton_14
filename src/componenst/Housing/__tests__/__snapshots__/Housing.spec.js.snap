// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Housing component should match snapshot 1`] = `
<Fragment>
  <ForwardRef
    key="1"
    onPress={[Function]}
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://cde.3.elcomercio.pe/ima/0/1/7/2/7/1727988.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="2"
    onPress={[Function]}
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://elcomercio.pe/resizer/Ioy5PRa28UDtkHbulEPQGkZo8Xk=/980x0/smart/filters:format(jpeg):quality(75)/cloudfront-us-east-1.images.arcpublishing.com/elcomercio/2DJTINJ2QZF4HDZ6PUH2CMAZPA.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="3"
    onPress={[Function]}
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://turismo.org.pe/wp-content/uploads/2018/12/lima-peru.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </View>
  </ForwardRef>
</Fragment>
`;
